packages/nextjs-template/context/useDendronContext.tsx

```diff
@@ -0,0 +1,8 @@
+import { useContext } from "react";
+import { dendronContext } from "./DendronProvider";
+
+export function useDendronContext() {
```

Comment:
not against providing our own context but pros/cons of using this vs a redux slice?

Reply:
Basic Use Context:
Pros
+ Simple
+ TypeScript Friendly
+ Can easily switch to Redux if state management becomes more tedious

Cons
- Can't use Redux middleware


Redux Slice:
Pros:
+ Redux middlewares can sometimes provide extra benefit

Cons:
- Lots of boilerplate
- Redux requires a lot of extra TypeScript